wallet-new/src/Cardano/Wallet/API/V1/Migration/Types.hs

```diff
@@ -60,16 +62,40 @@ instance (Migrate from to, Typeable from, Typeable to) => Migrate [from] (NonEmp
         ]
     eitherMigrate (x:xs) = (:|) <$> eitherMigrate x <*> mapM eitherMigrate xs
 
-instance Migrate (V0.CWallet, V0.WalletInfo) V1.Wallet where
-    eitherMigrate (V0.CWallet{..}, V0.WalletInfo{..}) =
+instance Migrate (V0.CWallet, OldStorage.WalletInfo, Maybe Core.ChainDifficulty) V1.Wallet where
+    eitherMigrate (V0.CWallet{..}, OldStorage.WalletInfo{..}, currentBlockchainDepth) =
         V1.Wallet <$> eitherMigrate cwId
                   <*> pure (V0.cwName cwMeta)
                   <*> eitherMigrate cwAmount
                   <*> pure cwHasPassphrase
                   <*> eitherMigrate cwPassphraseLU
                   <*> eitherMigrate _wiCreationTime
                   <*> eitherMigrate (V0.cwAssurance _wiMeta)
-
+                  <*> eitherMigrate (_wiSyncState, _wiSyncStatistics, currentBlockchainDepth)
+
+instance Migrate (OldStorage.WalletSyncState, OldStorage.SyncStatistics, Maybe Core.ChainDifficulty) V1.SyncState where
+    eitherMigrate (wss, stats, currentBlockchainDepth) =
+        case wss of
+            OldStorage.NotSynced         -> V1.Restoring <$> eitherMigrate (stats, currentBlockchainDepth)
+            OldStorage.RestoringFrom _ _ -> V1.Restoring <$> eitherMigrate (stats, currentBlockchainDepth)
+            OldStorage.SyncedWith _      -> pure V1.Synced
+
+instance Migrate (OldStorage.SyncStatistics, Maybe Core.ChainDifficulty) V1.SyncProgress where
+    eitherMigrate (OldStorage.SyncStatistics{..}, currentBlockchainDepth) =
+        let unknownCompletionTime = Core.Timestamp $ fromMicroseconds (fromIntegral (maxBound :: Int))
+            percentage = case currentBlockchainDepth of
+                Nothing -> 0.0
+                Just nd | wspCurrentBlockchainDepth >= nd -> 100
+                Just nd -> (fromIntegral wspCurrentBlockchainDepth / max 1.0 (fromIntegral nd)) * 100.0
+            toMs (Core.Timestamp microsecs) =
+              V1.mkEstimatedCompletionTime (round @Double $ (realToFrac (toMicroseconds microsecs) / 1000.0))
```

Comment:
The acute reader will notice that `mkEstimatedCompletionTime` has `Word` precision whereas we are downcasting from an `Integer`, and thus we could surely overflow. In practice, if such a thing would happen it would mean our `EstimatedCompletionTime` is obscenely high (we are talking about roughly `584942417` years if I did the math correctly 😅 ).